ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -110,8 +110,9 @@
         offset = new Offset(logSegmentName, value.getLong());
         flags = value.get();
         long expiresAt = value.getInt();
-        expiresAtMs = expiresAt >= 0 ? TimeUnit.SECONDS.toMillis(expiresAt) : Utils.Infinite_Time;
-        originalMessageOffset = value.getLong();
+        expiresAtMs = expiresAtMs != Utils.Infinite_Time && expiresAt >= 0 ? TimeUnit.SECONDS.toMillis(expiresAt)
```

Comment:
You are using `expiresAtMs` instead of `expiresAt` in the first part of the check. None of the tests caught this?
IntelliJ can warn you about this. In the 2017 version, you could go to "Preferences -> Editor -> Inspections -> Initialization Issues" and check "Instance field used before initialization".

Reply:
Ah thanks ! Will switch that on.

As to tests catching this - they can't since `expiresAt >= 0` is `false` for `Utils.Infinite_Time` (which happens to be -1).

Reply:
Because as far as the code goes, the value of `Infinite_Time` is opaque to it (could be -1, could be Integer.MAX_VALUE, or something else)

Reply:
Fixed the original issue